sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/SymbolicExecution/CSharp/CSharpExplodedGraph.cs

```diff
@@ -40,16 +40,21 @@ internal class CSharpExplodedGraph : AbstractExplodedGraph
     {
         private const string isNullOrEmpty = "IsNullOrEmpty";
         private const string isNullOrWhiteSpace = "IsNullOrWhiteSpace";
+        private readonly NullPointerDereference.NullPointerCheck nullPointerCheck;
 
         public CSharpExplodedGraph(IControlFlowGraph cfg, ISymbol declaration, SemanticModel semanticModel, AbstractLiveVariableAnalysis lva)
             : base(cfg, declaration, semanticModel, lva)
         {
+            nullPointerCheck = new NullPointerDereference.NullPointerCheck(this);
```

Comment:
`NullPointerCheck` is added by default by the `CSharpExplodedGraph` in order to stop the visit early when a null pointer dereference is found.

This check was previously replaced by `PublicMethodArgumentsShouldBeCheckedForNull` and `NullPointerDereference`.

In order to be able to run all the rules within one single semantic execution pass we have to reuse instead of replacing the check.